fetch/api/redirect/redirect-method.any.js

```diff
@@ -8,25 +8,37 @@
 // server is expected to echo the request body. The default is the empty string
 // if the request after redirection isn't POST; otherwise it's |opts.body|.
 function redirectMethod(desc, redirectUrl, redirectLocation, redirectStatus, method, expectedMethod, opts) {
-  var url = redirectUrl;
-  var urlParameters = "?redirect_status=" + redirectStatus;
+  let url = redirectUrl;
+  let urlParameters = "?redirect_status=" + redirectStatus;
   urlParameters += "&location=" + encodeURIComponent(redirectLocation);
 
-  var requestInit = {"method": method, "redirect": "follow"};
+  let requestInit = {"method": method, "redirect": "follow"};
   opts = opts || {};
-  if (opts.body)
+  if (opts.body) {
     requestInit.body = opts.body;
+  }
 
   promise_test(function(test) {
     return fetch(url + urlParameters, requestInit).then(function(resp) {
+      let isPostEnded = (expectedMethod == "POST");
```

Comment:
It only ends if it's a 301/302/303, right?

Reply:
Yes, `expectedMethod == "POST"` implies it's 301/302/303